hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/TestCreateTable.scala

```diff
@@ -360,6 +381,84 @@ class TestCreateTable extends TestHoodieSqlBase {
     }
   }
 
+  test("Test Create Table As Select With Tblproperties For Filter Props") {
+    Seq("cow", "mor").foreach { tableType =>
+      val tableName = generateTableName
+      spark.sql(
+        s"""
```

Comment:
This SQL makes me confused.
Should the table identifier be consistent with the property of "hoodie.database.name" ?

Reply:
> This SQL makes me confused. Should the table identifier be consistent with the property of "hoodie.database.name" ?
> 
> I mean the CTAS should start with "create table databaseName.$tableName"
It doesn't have to be set. It has a default value in `HoodieCatalogTable`

```scala
val hoodieDatabaseName = formatName(spark, table.identifier.database.getOrElse(spark.sessionState.catalog.getCurrentDatabase))
      HoodieTableMetaClient.withPropertyBuilder()
        .fromProperties(properties)
        .setDatabaseName(hoodieDatabaseName)
```

Reply:
`hoodie.database.name` is consistent with the table identifier, not the table identifier consistent with `hoodie.database.name`